src/frontends/tensorflow/src/tf_utils.cpp

```diff
@@ -190,3 +208,146 @@ ov::Any ov::frontend::tensorflow::unpack_tensor_proto(const ::tensorflow::Tensor
+                                                          const ov::frontend::NamedOutputVector& ov_outputs,
+                                                          const std::set<ov::Output<ov::Node>>& input_control_deps,
+                                                          std::set<ov::Output<ov::Node>>& output_control_deps) {
+    // returns if there is conditional flow to propagate
+    // it checks all producer-nodes connected via data edges and control dependencies
+
+    // compute combined markers
+    // it is a map from conditional flow marker to a set of branch markers
+    std::unordered_map<uint32_t, std::unordered_set<uint32_t>> combined_markers;
+    for (const auto& ov_input : ov_inputs) {
+        // walk through all data producer and collect conditional flow markers
+        const shared_ptr<const Node>& producer_node = ov_input.get_node_shared_ptr();
+        uint32_t branch_index = static_cast<uint32_t>(ov_input.get_index());
+        if (!ov::frontend::tensorflow::cf_marker_exists(producer_node)) {
+            continue;
+        }
+        auto producer_markers = ov::frontend::tensorflow::get_cf_marker(producer_node);
+        copy_conditional_flow_marker(producer_markers, combined_markers);
+        // if data goes from Switch node, it needs to create a new marker and branch marker
+        for (auto new_marker : producer_markers.new_markers) {
+            if (combined_markers.count(new_marker) > 0) {
+                // combine two sets of branch markers
+                combined_markers[new_marker].insert(branch_index);
+            } else {
+                combined_markers[new_marker] = {branch_index};
+            }
+        }
+    }
+    for (const auto& input_control_dep : input_control_deps) {
+        // walk through all control dependencies and collect conditional flow markers
+        const shared_ptr<const Node>& producer_node = input_control_dep.get_node_shared_ptr();
+        uint32_t branch_index = static_cast<uint32_t>(input_control_dep.get_index());
+        if (!ov::frontend::tensorflow::cf_marker_exists(producer_node)) {
+            continue;
+        }
+        auto producer_markers = ov::frontend::tensorflow::get_cf_marker(producer_node);
+        copy_conditional_flow_marker(producer_markers, combined_markers);
+        // if data goes from Switch node, it needs to create a new marker and branch marker
+        for (auto new_marker : producer_markers.new_markers) {
+            if (combined_markers.count(new_marker) > 0) {
+                // combine two sets of branch markers
+                combined_markers[new_marker].insert(branch_index);
+            } else {
+                combined_markers[new_marker] = {branch_index};
+            }
+        }
+    }
+
+    // walk through all nodes and mark them if needed
+    bool to_propagate = false;
+    for (const auto& ov_output : ov_outputs) {
+        const auto& node = ov_output.port.get_node_shared_ptr();
```

Comment:
Curious: I'm not sure I understand this style. Sometime you use shared_ptr<ov::Node> (line 271), sometime - auto. Does it mean something?

Reply:
More often I use `auto` just to avoid long types in the code. But sometimes I need `const shared_pt<const Node>` for which I cannot use `auto` but some OV API requires `const` object pointed by `shared_ptr`